packages/core/src/__tests__/unit/application-lifecycle.unit.ts

```diff
@@ -22,6 +22,71 @@ import {
 } from '../..';
 
 describe('Application life cycle', () => {
+  describe('state', () => {
+    it('updates application state', async () => {
+      const app = new Application();
+      expect(app.state).to.equal('created');
+      const start = app.start();
+      expect(app.state).to.equal('starting');
+      await start;
+      expect(app.state).to.equal('started');
+      const stop = app.stop();
+      expect(app.state).to.equal('stopping');
+      await stop;
+      expect(app.state).to.equal('stopped');
+    });
```

Comment:
This test ignores `booting` and `booted` states. Please verify that boot + start work together as intended & expected.